### PR TITLE
GH-87 Finish sending logic

### LINKED ISSUE #87: Parcel sending logic
- [x] Finish ParcelItemStorageGUI *requires testing*
- [x] Touch-up other GUIs to make sure they are fully-functional
- [x] Finish ParcelSendingGUI and its logic *(partially done)*
- [ ] Make sure the receiver can redeem the parcel
- [ ] Update the ParcelStatus in real-time
- [ ] Create a task system that will save parcel's remaining time in database

Potentially:
- #66 
- #63

### PR DESCRIPTION
**Pull request by Jakubk15**
Resolves #87

### CODE REVIEW

**Comment by Jakubk15** [comment]
GsonItemSerializer requires work to properly serialize data in database

**Comment by Jakubk15** [comment]
```
[20:04:44 ERROR]: Could not pass event InventoryCloseEvent to ParcelLockers v1.0.0-SNAPSHOT
java.lang.ExceptionInInitializerError: null
	at com.eternalcode.parcellockers.parcel.gui.ParcelItemStorageGUI.lambda$show$4(ParcelItemStorageGUI.java:86) ~[ParcelLockers v1.0.0-SNAPSHOT (MC 1.8.8-1.20.x).jar:?]
	at dev.triumphteam.gui.guis.GuiListener.onGuiClose(GuiListener.java:138) ~[ParcelLockers v1.0.0-SNAPSHOT (MC 1.8.8-1.20.x).jar:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor61.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77) ~[paper-api-1.20.2-R0.1-SNAPSHOT.jar:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:81) ~[paper-api-1.20.2-R0.1-SNAPSHOT.jar:git-Paper-318]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[paper-api-1.20.2-R0.1-SNAPSHOT.jar:?]
	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[paper-1.20.2.jar:git-Paper-318]
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126) ~[paper-1.20.2.jar:git-Paper-318]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615) ~[paper-api-1.20.2-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_20_R2.event.CraftEventFactory.handleInventoryCloseEvent(CraftEventFactory.java:1680) ~[paper-1.20.2.jar:git-Paper-318]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleContainerClose(ServerGamePacketListenerImpl.java:2866) ~[?:?]
	at org.bukkit.craftbukkit.v1_20_R2.event.CraftEventFactory.callInventoryOpenEventWithTitle(CraftEventFactory.java:1472) ~[paper-1.20.2.jar:git-Paper-318]
	at org.bukkit.craftbukkit.v1_20_R2.event.CraftEventFactory.callInventoryOpenEventWithTitle(CraftEventFactory.java:1460) ~[paper-1.20.2.jar:git-Paper-318]
	at org.bukkit.craftbukkit.v1_20_R2.entity.CraftHumanEntity.openCustomInventory(CraftHumanEntity.java:361) ~[paper-1.20.2.jar:git-Paper-318]
	at org.bukkit.craftbukkit.v1_20_R2.entity.CraftHumanEntity.openInventory(CraftHumanEntity.java:345) ~[paper-1.20.2.jar:git-Paper-318]
	at dev.triumphteam.gui.guis.BaseGui.open(BaseGui.java:422) ~[ParcelLockers v1.0.0-SNAPSHOT (MC 1.8.8-1.20.x).jar:?]
	at com.eternalcode.parcellockers.parcel.gui.ParcelSendingGUI.show(ParcelSendingGUI.java:83) ~[ParcelLockers v1.0.0-SNAPSHOT (MC 1.8.8-1.20.x).jar:?]
	at com.eternalcode.parcellockers.parcel.gui.ParcelItemStorageGUI.lambda$show$1(ParcelItemStorageGUI.java:45) ~[ParcelLockers v1.0.0-SNAPSHOT (MC 1.8.8-1.20.x).jar:?]
	at dev.triumphteam.gui.guis.GuiListener.onGuiClick(GuiListener.java:102) ~[ParcelLockers v1.0.0-SNAPSHOT (MC 1.8.8-1.20.x).jar:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor63.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77) ~[paper-api-1.20.2-R0.1-SNAPSHOT.jar:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:81) ~[paper-api-1.20.2-R0.1-SNAPSHOT.jar:git-Paper-318]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[paper-api-1.20.2-R0.1-SNAPSHOT.jar:?]
	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[paper-1.20.2.jar:git-Paper-318]
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126) ~[paper-1.20.2.jar:git-Paper-318]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615) ~[paper-api-1.20.2-R0.1-SNAPSHOT.jar:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleContainerClick(ServerGamePacketListenerImpl.java:3108) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:58) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundContainerClickPacket.handle(ServerboundContainerClickPacket.java:23) ~[?:?]
	at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:53) ~[?:?]
	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.20.2.jar:git-Paper-318]
	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1324) ~[paper-1.20.2.jar:git-Paper-318]
	at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:193) ~[paper-1.20.2.jar:git-Paper-318]
	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
	at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1301) ~[paper-1.20.2.jar:git-Paper-318]
	at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1294) ~[paper-1.20.2.jar:git-Paper-318]
	at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136) ~[?:?]
	at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1272) ~[paper-1.20.2.jar:git-Paper-318]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1160) ~[paper-1.20.2.jar:git-Paper-318]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:315) ~[paper-1.20.2.jar:git-Paper-318]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.IllegalArgumentException
Caused by: java.lang.IllegalArgumentException

	at com.google.gson.internal.$Gson$Preconditions.checkArgument($Gson$Preconditions.java:54) ~[gson-2.10.1.jar:?]
	at com.google.gson.GsonBuilder.registerTypeAdapter(GsonBuilder.java:606) ~[gson-2.10.1.jar:?]
	at com.eternalcode.parcellockers.util.ItemUtil.<clinit>(ItemUtil.java:16) ~[ParcelLockers v1.0.0-SNAPSHOT (MC 1.8.8-1.20.x).jar:?]
	... 43 more
	```

File: README.md [hunk 1]
```diff
@@ -1,84 +1,89 @@
...
+# Welcome to ParcelLockers! 🚀
+
+ParcelLockers is a minecraft plugin, dedicated for Spigot-based servers.. 💎 Crafted with care, ParcelLockers
+is dedicated to incorporating highly practical server function of transfering items across the Minecraft world,
+especially on long distances.
+
+## ℹ️ Information
+
+- ParcelLockers fully supports Minecraft's latest minor versions starting from each major version, starting from 1.17
+  onward, e.g. `1.17.1`, `1.18.2`, `1.19.4`, `1.20.6`.
```
Comment (Jakubk15) [inline README.md:26]:
```suggestion
  onward, e.g. `1.17.1`, `1.18.2`, `1.19.4`, `1.20.6`, `1.21.1`.
```

File: README.md [hunk 2]
```diff
@@ -1,84 +1,89 @@
...
+- If you have any questions, you can ask us about it on our [Discord server](https://discord.gg/FQ7jmGBd6c).
+
+## ❗Warning
 
 ParcelLockers is at an very early stage of development, and you use it at your own risk.
 
-# Building
+## 🏗️ Compiling JAR from Source
+
+To build ParcelLockers, follow these steps (Make sure you have **JDK 17 or higher**):
```
Comment (Jakubk15) [inline README.md:37]:
```suggestion
To build ParcelLockers, follow these steps (Make sure you have **JDK 21 or higher**):
```

File: README.md [hunk 3]
```diff
@@ -88,13 +93,23 @@ more.
 - [Spigot API](https://www.spigotmc.org/wiki/spigot-maven/#build-gradle)
 - [expressible](https://github.com/dzikoysk/expressible)
 - [panda-utilities](https://github.com/panda-lang/panda)
-- [Errorprone](https://github.com/google/error-prone)
 - [TriumphGUI](https://github.com/TriumphTeam/triumph-gui)
 - [bStats](https://bstats.org/)
+- [jackson-bukkit](https://github.com/eldoriarpg/jackson-bukkit)
```
Comment (Jakubk15) [inline README.md:98]:
```suggestion
- [jackson-bukkit](https://github.com/eldoriarpg/jackson-bukkit)
- [SignGUI](https://github.com/Rapha149/SignGUI)
```

File: src/main/java/com/eternalcode/parcellockers/gui/implementation/locker/ParcelSendingGUIState.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    private String parcelName;
+    private String parcelDescription;
+    private ParcelSize size;
+    private UUID receiver;
+    private boolean priority;
+    private UUID entryLocker;
+    private UUID destinationLocker;
+
+
+    public ParcelSendingGUIState() {
```
Comment (Jakubk15) [inline src/main/java/com/eternalcode/parcellockers/gui/implementation/locker/ParcelSendingGUIState.java:18]:
```suggestion
    private UUID destinationLocker;

    public ParcelSendingGUIState() {
```

File: src/main/java/com/eternalcode/parcellockers/gui/implementation/locker/ReceiverSelectionGui.java [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+import org.bukkit.entity.Player;
+import org.bukkit.plugin.Plugin;
+import org.bukkit.scheduler.BukkitScheduler;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
+
+// TODO: Maybe in future cache this globally, so props wont change after gui exit, because new instance is created.
```
Comment (Jakubk15) [inline src/main/java/com/eternalcode/parcellockers/gui/implementation/locker/ReceiverSelectionGui.java:29]:
```suggestion
```

File: src/main/java/com/eternalcode/parcellockers/util/legacy/LegacyColorProcessor.java [hunk 6]
```diff
@@ -10,6 +10,6 @@ public class LegacyColorProcessor implements UnaryOperator<Component> {
     @Override
     public Component apply(Component component) {
         return component.replaceText(builder -> builder.match(Pattern.compile(".*"))
-                .replacement((matchResult, builder1) -> Legacy.component(matchResult.group())));
+            .replacement((matchResult, builder1) -> Legacy.component(matchResult.group())));
     }
 }
```
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/util/legacy/LegacyColorProcessor.java:15]: Performance issue, replace by fixed eternalcode-commons-adventure.

File: src/main/java/com/eternalcode/parcellockers/util/legacy/Legacy.java
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/util/legacy/Legacy.java:1]: Replace with eternalcode-commons-adventure

File: src/main/java/com/eternalcode/parcellockers/util/RandomUtil.java [hunk 7]
```diff
@@ -16,4 +18,22 @@ public static String generateDeliveryCode() {
...
+    public static String randomParcelDescription() {
+        int leftLimit = 97; // letter 'a'
+        int rightLimit = 122; // letter 'z'
+        int targetStringLength = 16;
+
+        return RANDOM.ints(leftLimit, rightLimit + 1)
+            .limit(targetStringLength)
+            .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
+            .toString();
+    }
```
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/util/RandomUtil.java:37]: Maybe ParcelUtil?

File: src/test/java/com/eternalcode/parcellockers/database/ParcelDatabaseServiceIntegrationTest.java [hunk 8]
```diff
@@ -29,14 +31,14 @@ class ParcelDatabaseServiceIntegrationTest extends ParcelLockerIntegrationSpec {
     @Test
     void test() {
         HikariDataSource dataSource = buildHikariDataSource(mySQLContainer);
-        ParcelRepositoryImpl parcelRepositoryImpl = new ParcelRepositoryImpl(dataSource);
+        ParcelRepository ParcelRepository = new ParcelRepositoryImpl(dataSource);
```
Comment (vLuckyyy) [inline src/test/java/com/eternalcode/parcellockers/database/ParcelDatabaseServiceIntegrationTest.java:34]: use camelCase

File: src/main/java/com/eternalcode/parcellockers/util/ItemUtil.java [hunk 9]
```diff
@@ -1,42 +1,135 @@
...
+        for (String line : lore) {
+            newLore.add(formatter.format(line));
+        }
+
+        return newLore;
     }
 
-    public static ItemStack stringToItemStack(String string) {
-        return GSON.fromJson(string, ItemStack.class);
+    @ApiStatus.Internal
```
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/util/ItemUtil.java:126]: You don't need annotations here, it's not an api.

File: src/main/java/com/eternalcode/parcellockers/util/ItemUtil.java [hunk 10]
```diff
@@ -1,42 +1,135 @@
...
+
+    public static ItemStack deserialize(String string) {
+        try {
+            return JSON.readValue(string, ItemStack.class);
         }
-        
-        ItemMeta secondMeta = second.getItemMeta();
-        if (secondMeta == null) {
-            return false;
+        catch (JsonProcessingException e) {
```
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/util/ItemUtil.java:63]: I would prefer if you used full names everywhere in the plugin. `exception`.

File: src/main/java/com/eternalcode/parcellockers/util/ItemUtil.java [hunk 11]
```diff
@@ -1,42 +1,135 @@
...
+        }
+
+        List<String> newLore = new ArrayList<>();
+
+        for (String line : lore) {
+            newLore.add(formatter.format(line));
+        }
+
+        return newLore;
     }
```
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/util/ItemUtil.java:124]: Pomyślalem że mógłbyś zrobić to w Paczce Parcel jako `ParcelPlaceholderUtil`

File: src/main/java/com/eternalcode/parcellockers/util/ItemUtil.java [hunk 12]
```diff
@@ -1,42 +1,135 @@
...
+import panda.utilities.text.Formatter;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
 
 public class ItemUtil {
```
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/util/ItemUtil.java:26]: ItemSerializer, ItemSerializationUtil maybe?

File: src/main/java/com/eternalcode/parcellockers/util/InventoryUtil.java [hunk 13]
```diff
@@ -0,0 +1,19 @@
+package com.eternalcode.parcellockers.util;
+
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+
+public class InventoryUtil {
+
+    public static void addItem(Player player, ItemStack itemStack) {
```
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/util/InventoryUtil.java:9]: When you rework it on eternalcode-commons, you can use the itemutil that is built in there.

File: src/main/java/com/eternalcode/parcellockers/util/DurationUtil.java [hunk 14]
```diff
@@ -18,8 +18,8 @@ public static String format(Duration duration, boolean removeMillis) {
         }
 
         return duration.toString()
```
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/util/DurationUtil.java:20]: When you convert to eternalcode-commons, you can use the time formatting that is added there.

File: src/main/java/com/eternalcode/parcellockers/user/UserRepository.java [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+
+    CompletableFuture<Optional<User>> getUser(UUID uuid);
+
+    CompletableFuture<Optional<User>> getUser(String name);
+
+    CompletableFuture<Void> save(User user);
+
+    CompletableFuture<Void> changeName(UUID uuid, String newName);
+
+    CompletableFuture<UserPageResult> findPage(Page page);
```
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/user/UserRepository.java:19]: You use the keyword `get` everywhere, use it here too.

**Review by vLuckyyy — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS [hunk 16]
```diff
@@ -1 +1 @@
-* @EternalCodeTeam/eternalcodeteam-maintainers @Jakubk15
\ No newline at end of file
+* @EternalCodeTeam/team @Jakubk15
```
Comment (vLuckyyy) [inline .github/CODEOWNERS:1]: unrelated to pr title

File: README.md
Comment (vLuckyyy) [inline README.md:1]: unrelated readme changes to pr title.

File: assets/ParcelLockers.svg
Comment (vLuckyyy) [inline assets/ParcelLockers.svg:1]: unrelated to pr title

File: src/main/java/com/eternalcode/parcellockers/ParcelLockers.java [hunk 17]
```diff
@@ -84,57 +101,75 @@ public void onEnable() {
...
+            .invalidUsage(new InvalidUsageImpl(announcer, config))
+            .missingPermission(new PermissionMessage(announcer, config))
+            .build();
+
+        /*if (!this.setupEconomy()) {
+            this.getLogger().severe("Disabling due to no Vault dependency or its implementator(s) found!");
             server.getPluginManager().disablePlugin(this);
             return;
-        }
+        }*/
```
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/ParcelLockers.java:143]: Unrelated to pr title.

File: src/main/java/com/eternalcode/parcellockers/ParcelLockersCommand.java [hunk 18]
```diff
@@ -3,42 +3,39 @@
...
+import dev.rollczi.litecommands.annotations.command.Command;
+import dev.rollczi.litecommands.annotations.context.Context;
+import dev.rollczi.litecommands.annotations.execute.Execute;
+import dev.rollczi.litecommands.annotations.permission.Permission;
 import org.bukkit.command.CommandSender;
 import org.bukkit.entity.Player;
 import org.bukkit.inventory.ItemStack;
 
-@Route(name = "parcellockers", aliases = {"parcellocker"})
+@Command(name = "parcellockers", aliases = { "parcellocker" })
```
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/ParcelLockersCommand.java:15]: Why you change major version of lite commands in pull requests that mean "finish sending logic"

File: src/main/java/com/eternalcode/parcellockers/command/handler/InvalidUsageImpl.java
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/command/handler/InvalidUsageImpl.java:1]: unrelated to pr title.

File: src/main/java/com/eternalcode/parcellockers/command/handler/PermissionMessage.java
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/command/handler/PermissionMessage.java:1]: unrelated to pr title.

File: src/main/java/com/eternalcode/parcellockers/configuration/ConfigurationManager.java
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/configuration/ConfigurationManager.java:1]: unrelated code-style changes

File: src/main/java/com/eternalcode/parcellockers/configuration/implementation/ConfigItem.java [hunk 19]
```diff
@@ -51,8 +90,24 @@ public ConfigItem setLore(List<String> lore) {
...
         return this;
     }
+
+    @Override
+    public ConfigItem clone() {
+        try {
+            ConfigItem cloned = (ConfigItem) super.clone();
+            cloned.lore = new ArrayList<>(this.lore); // Deep copy of the list
+            return cloned;
+        } catch (CloneNotSupportedException e) {
```
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/configuration/implementation/ConfigItem.java:109]: in many places of the code you improve the style of the code from the main branch, but here you have not changed it anymore?

File: src/main/java/com/eternalcode/parcellockers/conversation/ParcelLockerPlacePrompt.java [hunk 20]
```diff
@@ -28,8 +24,7 @@ public Prompt acceptInput(@NotNull ConversationContext context, @Nullable String
     @NotNull
     @Override
     public String getPromptText(@NotNull ConversationContext context) {
-        this.announcer.sendMessage((CommandSender) context.getForWhom(), this.config.messages.enterDescriptionPrompt);
-        return StringUtils.EMPTY;
+        return ChatColor.translateAlternateColorCodes('&', this.config.messages.enterDescriptionPrompt);
```
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/conversation/ParcelLockerPlacePrompt.java:27]: Dlaczego to akurat wykorzystuje `translateAlternateColorCodes`?

File: src/main/java/com/eternalcode/parcellockers/gui/implementation/locker/DestinationSelectionGUI.java [hunk 21]
```diff
@@ -0,0 +1,135 @@
...
+
+    public DestinationSelectionGUI(Plugin plugin, BukkitScheduler scheduler, PluginConfiguration config, MiniMessage miniMessage, LockerRepository lockerRepository, ParcelSendingGUI sendingGUI, ParcelSendingGUIState state) {
+        this.plugin = plugin;
+        this.scheduler = scheduler;
+        this.config = config;
+        this.miniMessage = miniMessage;
+        this.lockerRepository = lockerRepository;
+        this.sendingGUI = sendingGUI;
+        this.state = state;
+    }
```
Comment (vLuckyyy) [inline src/main/java/com/eternalcode/parcellockers/gui/implementation/locker/DestinationSelectionGUI.java:48]: It is probably possible to divide it so that you have less dependence here.

**Review by vLuckyyy — CHANGES_REQUESTED** [review]

File: src/main/java/com/eternalcode/parcellockers/conversation/ParcelLockerPlacePrompt.java (hunk 20 above)
Comment (Jakubk15) [reply src/main/java/com/eternalcode/parcellockers/conversation/ParcelLockerPlacePrompt.java:27]: Bo to funkcja Spigotowa która nie wspiera Adventure API

File: src/main/java/com/eternalcode/parcellockers/gui/implementation/locker/DestinationSelectionGUI.java (hunk 21 above)
Comment (Jakubk15) [reply src/main/java/com/eternalcode/parcellockers/gui/implementation/locker/DestinationSelectionGUI.java:48]: I already shrinked this by 50% by isolating dependencies into the `ParcelSendingGUIState`.

File: src/main/java/com/eternalcode/parcellockers/util/DurationUtil.java (hunk 14 above)
Comment (Jakubk15) [reply src/main/java/com/eternalcode/parcellockers/util/DurationUtil.java:20]: Will do in another, related PR.

File: src/main/java/com/eternalcode/parcellockers/command/handler/InvalidUsageImpl.java [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+
+    public InvalidUsageImpl(NotificationAnnouncer announcer, PluginConfiguration config) {
+        this.announcer = announcer;
+        this.config = config;
+    }
+
+    @Override
+    public void handle(Invocation<CommandSender> invocation, InvalidUsage<CommandSender> result, ResultHandlerChain<CommandSender> chain) {
+        Schematic schematic = result.getSchematic();
+        List<String> schematics = schematic.all();
```
Comment (imDMK) [inline src/main/java/com/eternalcode/parcellockers/command/handler/InvalidUsageImpl.java:28]: Move it after the if to make the code more readable

File: src/main/java/com/eternalcode/parcellockers/gui/implementation/remote/ParcelListGUI.java [hunk 23]
```diff
@@ -112,38 +114,47 @@ private void show(Player player, Page page) {
...
-        UUID destinationLocker = parcel.destinationLocker();
-
-        this.lockerRepository.findLocker(destinationLocker).ifPresent(locker -> formatter
-            .register("{POSITION_X}", locker.position().x())
-            .register("{POSITION_Y}", locker.position().y())
-            .register("{POSITION_Z}", locker.position().z())
-        );
+            .register("{RECIPIENTS}", recipients.toString());
+
+        Optional<Locker> lockerOptional = this.lockerRepository.findByUUID(parcel.destinationLocker()).join();
```
Comment (imDMK) [inline src/main/java/com/eternalcode/parcellockers/gui/implementation/remote/ParcelListGUI.java:145]: Blocks the entire thread. Use thenApply

File: src/main/java/com/eternalcode/parcellockers/locker/controller/LockerPlaceController.java [hunk 24]
```diff
@@ -17,26 +15,49 @@
...
+
+        if (secondMeta == null) {
+            return false;
+        }
+
+        if (first.getType() != second.getType()) {
+            return false;
+        }
+
+        return new HashSet<>(firstMeta.getLore()).containsAll(secondMeta.getLore())
```
Comment (imDMK) [inline src/main/java/com/eternalcode/parcellockers/locker/controller/LockerPlaceController.java:57]: Isn't it better to just use the compareTo method at this point?

File: src/main/java/com/eternalcode/parcellockers/parcel/command/argument/ParcelLockerArgument.java [hunk 25]
```diff
@@ -0,0 +1,38 @@
...
+            .collect(SuggestionResult.collector());
+    }
+
+    @Override
+    protected ParseResult<Locker> parse(Invocation<CommandSender> invocation, Argument<Locker> argument, String s) {
+        return ParseResult.success(this.lockerRepository.cache().values()
+            .stream()
+            .filter(locker -> locker.description().equalsIgnoreCase(s))
+            .findFirst()
+            .orElse(null));
```
Comment (imDMK) [inline src/main/java/com/eternalcode/parcellockers/parcel/command/argument/ParcelLockerArgument.java:36]: Czy zwracanie null w tym miejscu nie wyrzuci błędu przez litecommands?

File: src/main/java/com/eternalcode/parcellockers/gui/implementation/remote/ParcelListGUI.java (hunk 23 above)
Comment (Jakubk15) [reply src/main/java/com/eternalcode/parcellockers/gui/implementation/remote/ParcelListGUI.java:145]: Blocking is allowed on async threads.

File: src/main/java/com/eternalcode/parcellockers/locker/controller/LockerPlaceController.java (hunk 24 above)
Comment (Jakubk15) [reply src/main/java/com/eternalcode/parcellockers/locker/controller/LockerPlaceController.java:57]: `equals` method is more intuitive and directly checks for equality.
`compareTo` is generally used for ordering and might be less clear in this context.

Comment (imDMK) [reply src/main/java/com/eternalcode/parcellockers/locker/controller/LockerPlaceController.java:57]: No. The current way you compare the name and lore of items creates a new hashset each time, which can be inefficient. The compareTo method compares the items directly (with enchants etc) which is a better solution. Let the rest of the members comment on this.

**Review by Rollczi — CHANGES_REQUESTED** [review]

**Review by P1otrulla — DISMISSED** [review]
Chyba nic nie wpadło mi w oczy, ale przy duży ten PR

File: src/main/java/com/eternalcode/parcellockers/gui/implementation/locker/ParcelSendingGUI.java [hunk 26]
```diff
@@ -0,0 +1,353 @@
...
+                        this.announcer.sendMessage(player, settings.messages.parcelNameCannotBeEmpty);
+                        return Collections.emptyList();
+                    }
+
+                    this.state.setParcelName(name);
+                    this.announcer.sendMessage(player, settings.messages.parcelNameSet);
+
+                    List<String> lore = nameItem.lore;
+                    if (lore.size() > 1) {
+                        lore.remove(1);
```
[the remark this answers is not in the document]
Comment (Jakubk15) [reply src/main/java/com/eternalcode/parcellockers/gui/implementation/locker/ParcelSendingGUI.java:114]: Zobaczę w domu, z tego co widzę jest to ConfigItem który jest Cloneable

[the remark this answers is not in the document]
Comment (Rollczi) [reply src/main/java/com/eternalcode/parcellockers/gui/implementation/locker/ParcelSendingGUI.java:114]: Ten interface nic nie zmienia

**Review by igoyek — APPROVED** [review]
